build_defs/python.build_defs

```diff
@@ -744,12 +770,44 @@ def _wheel_entrypoint_binary(name:str, entrypoint:str, lib_rule, visibility, tes
         visibility = visibility,
     )
 
-def _interpreter_cmd(interpreter:str):
-    return f'$(out {interpreter})' if looks_like_build_label(interpreter) else interpreter
-
 
-def _interpreter_opts(opts:list):
-    return " ".join([f'--interpreter_options="{o}"' for o in opts])
+def _runtime_interpreter(build_interp:str, run_interps:list, run_args:list):
+    """Returns a string of please_pex options defining the run-time Python interpreter paths and
+    arguments to pass to the interpreter during invocation (falling back to default values where
+    necessary), and a list containing the names of the subset of those interpreters that are defined
+    as build targets.
+    """
+    opts = []
+    targets = []
+    for i in (run_interps or CONFIG.PYTHON.DEFAULT_RUNTIME_INTERPRETERS or [build_interp]):
+        if looks_like_build_label(i):
+            i_target = i.rpartition("|")[0] if "|" in i else i
+            # The location of a Python interpreter defined as a build target within the repo
+            # relative to the .pex output of a python_binary/python_test target that lists it as a
+            # run-time dependency varies depending on the context. If the target is running within a
+            # build environment (e.g. under `plz build` or `plz test`, either directly or as a
+            # dependency of some other target), Please will also copy the interpreter into the build
+            # directory if it is listed as a run-time dependency of the target. If the target is
+            # running outside a build environment (e.g. under `plz run`), Please will build the
+            # interpreter target, outputting it to plz-out/bin/. Relative to each of these paths,
+            # the interpreter will exist at the path given by passing its target name to the
+            # $(location) built-in function. However, this context is a run-time property of the
+            # .pex file; by prepending the interpreter path with "$PLZ_BIN_PATH/", the preamble will
+            # detect the context and substitute an appropriate path in place of $PLZ_BIN_PATH.
+            opts += [f"--interpreters='$PLZ_BIN_PATH/$(location {i})'"]
+            # If the target name ends with an entry point name, strip it - runtime_deps won't accept
+            # target names containing entry point names (which isn't really a problem for us,
+            # because the entire target will have to be built for the entry point to be provided
+            # anyway).
```

Comment:
Should this comment be attached to line 784? Or should line 784 be beneath this comment?